language-server/src/server.js

```diff
@@ -247,13 +251,26 @@ const validateSchema = async (textDocument) => {
       } else {
         diagnostics.push(buildDiagnostic(schemaInstance, "No dialect"));
       }
-
       continue;
     }
 
     try {
       const annotatedInstance = await annotate(dialectUri, schemaInstance);
 
+      const references = [];
+      for (const { keywordInstance } of getSemanticTokens(schemaResources)) {
```

Comment:
It's surprising to see `setSemanticTokens` here. I think I see what you're doing though. It's a hack to get locations that are keywords. I suppose it's fine for now because we don't have a better way, but please leave a comment explaining why this is used here.